### PR TITLE
fix: random error/prompt cleanups

### LINKED ISSUE #1332: cargo dist init in new js project gives strange prompts
repro:

1. add a dist-workspace toml to a js project
2. run cargo-dist init

```
ag_dubs@Ashleys-MBP-2 workers-sdk % vi dist-workspace.toml
ag_dubs@Ashleys-MBP-2 workers-sdk % cargo dist init
let's setup your cargo-dist config...

? looks like you deleted the cargo-dist-version key, add it back?
    this is the version of cargo-dist your releases should use
    (you're currently running 0.21.0) (y/n) › yes
```

### LINKED ISSUE #1336: would b nice to have the rich errors we have for toml parsing for json parsing
```
ag_dubs@Ashleys-MBP-2 workers-sdk % cargo dist build
  × We encountered an issue trying to read your workspace
  ├─▶ failed to parse JSON
  ╰─▶ trailing comma at line 42 column 2
 ```

### LINKED ISSUE #1338: Better errors/linting on command not found
I accidentally used the following for my custom build command

```toml
build-command = ["go build -o example ."]
```

This kept giving me the issue that the command could not be found, but was not telling me exactly what it was looking for. I got confused thinking I had to install Go somewhere specific.

The displayed error was;
```
building generic target (x86_64-apple-darwin via go build -o example .)
  × failed to exec generic build: go build -o example .
  ╰─▶ No such file or directory (os error 2)
```

Possibly suggestions:
- Make the error handling a bit more verbose
- Lint the build command before running it, if the user put a space in the first string, chances are it was this mistake?

### PR DESCRIPTION
**Pull request by Gankra**
* fixes #1332
* fixes #1336 
* fixes #1338 

Also does a general "don't say generic to users" pass

### CODE REVIEW

**Comment by Gankra** [comment]
the first commit has some stubbed out code I need to turn into Real Errors and not fake macros

File: cargo-dist/src/errors.rs [hunk 1]
```diff
@@ -480,6 +481,42 @@ pub enum DistError {
         /// The dist.toml/dist-workspace.toml that means it's ignored
         dist_manifest_path: Utf8PathBuf,
     },
+    /// Build command the looks like they put args in same string
```
Comment (mistydemeo) [inline cargo-dist/src/errors.rs:484]:
```suggestion
    /// Build command looks like they put args in same string
```

File: cargo-dist/src/errors.rs [hunk 2]
```diff
@@ -480,6 +481,42 @@ pub enum DistError {
...
+    #[error("package.json was missing a \"dist\" script\n{manifest}")]
+    #[diagnostic(help(
+        "https://opensource.axo.dev/cargo-dist/book/quickstart/javascript.html#adding-a-dist-script"
+    ))]
+    NoDistScript {
+        /// path to package.json
+        manifest: Utf8PathBuf,
+    },
+    /// missing "build-command" for a package that needs one
+
```
Comment (mistydemeo) [inline cargo-dist/src/errors.rs:503]: Extra newline here:

```suggestion
```

**Review by mistydemeo — APPROVED** [review]
Looks good, just some minor things I noticed.

**Pull request merged**